File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/events/GhostCarEvent.java
package com.bitfire.uracer.game.events;

public final class GhostCarEvent extends Event<GhostCarEvent.Type, GhostCarEvent.Order> {
    public GhostCarEvent() {
        super(Type.class, Order.class);
    }

    public enum Type {
        onGhostFadingOut, ReplayStarted, ReplayEnded
    }

    public enum Order {
        Default
    }

    public interface Listener extends Event.Listener<Type, Order> {
        @Override
        void handle(Object source, Type type, Order order);
    }
}

File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/events/GhostCarEvent.kt
package com.bitfire.uracer.game.events

class GhostCarEvent : Event<GhostCarEvent.Type, GhostCarEvent.Order>(Type::class.java, Order::class.java) {

    enum class Type {
        OnGhostFadingOut, ReplayStarted, ReplayEnded
    }

    enum class Order {
        Default
    }

    interface Listener : Event.Listener<Type, Order> {
        override fun handle(source: Any, type: Type, order: Order)
    }
}
